rust/storage/src/network_admission_control.rs

```diff
@@ -0,0 +1,146 @@
+use super::{GetError, Storage};
+use chroma_error::{ChromaError, ErrorCodes};
+use futures::{future::Shared, FutureExt, StreamExt};
+use parking_lot::Mutex;
+use std::{collections::HashMap, future::Future, pin::Pin, sync::Arc};
+use thiserror::Error;
+use tracing::{Instrument, Span};
+
+#[derive(Clone)]
+pub struct NetworkAdmissionControl {
```

Comment:
Maybe "AdmissionControlledStorage" is a more apt term here and we should think of this as a wrapper around Storage that performs admission control